flink-metrics/flink-metrics-jmx/src/main/java/org/apache/flink/metrics/jmx/JMXReporterFactory.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.metrics.jmx;
+
+import org.apache.flink.metrics.reporter.MetricReporterFactory;
+
+import java.util.Optional;
+import java.util.Properties;
+
+/**
+ * {@link MetricReporterFactory} for {@link JMXReporter}.
+ */
+public class JMXReporterFactory implements MetricReporterFactory {
+
+	static final String ARG_PORT = "port";
```

Comment:
This is out of scope of this PR but maybe something to consider for the future. I think the `MetricReporter` configuration parameters are hard to discover. Maybe we could use `ConfigOptions` and their utilities to bundle definition and description as well as to generate the reporter documentation automatically. We would also get type checks for free. One could, for example define a `JMXReporterConfigOptions` class which contains all the different `ConfigOptions`. Then only needs to include the generated config documentation in the `metrics.md` file.

Reply:
I brought up this very point when we added the InfluxDB reporter.

We cannot do this at the moment.
The docs generator enforces that all config option keys are well-defined; i.e., for a given key there is exactly 1 description and default value.

The parameters here are not fully qualified however but only suffixes, and would thus likely clash with other reporters that also have a "port" parameter.
I haven't found the time yet to add support for this.